### ISSUE TITLE
cmd/compile: Incorrect column number in build error for unused variable

### ISSUE THREAD

**Issue by bradleyfalzon** [golang/go#21317]
### What version of Go are you using (`go version`)?

```
go version go1.9rc1 linux/amd64
```

### What operating system and processor architecture are you using (`go env`)?

```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/bradleyf/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build365274238=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
```

### What did you do?

```
package main

import "fmt"

func main() {
        n, err := fmt.Print(1)
}
```

Run this program with `go build`

### What did you expect to see?

```
./main.go:6:2: n declared and not used
./main.go:6:5: err declared and not used
```

Note the column numbers.

### What did you see instead?

```
./main.go:6:22: n declared and not used
./main.go:6:22: err declared and not used
```

Column 22 for the above program is the argument to the `fmt.Print` function. There's nothing wrong with the argument, the problem is with the returned arguments.

It appears this may be a known issue, mentioned in https://github.com/golang/go/commit/2a5cf48f91c0a59eeb01b97ca6afaca311324206 but it's not clear whether this is the same issue.

At the very least, when the editor is highlighting the argument but the error is elsewhere on the same line, it's confusing.

**Comment by mvdan** [comment]
CC @mdempsky @griesemer

**Comment by bradleyfalzon** [comment]
Another example which I most likely the same issue but it is confusing when the editor is highlighting the wrong identifier:

```
package main

func main() {
        var a int64 = 1
        Foo("SOMEVAR", a)
}

func Foo(_ string, _ int) {}
```

```
$ go build
./main.go:5:6: cannot use a (type int64) as type int in argument to Foo
```

Column 6, in this case, is the first argument to the Foo function call, but the problem is with the second argument.

**Comment by griesemer** [comment]
Haven't checked if it's a duplicate but this can be improved for sure.

**Comment by shurcooL** [comment]
This is what a code editor that displays issues inline shows the user:

![image](https://user-images.githubusercontent.com/1924134/29301152-3a4c8ec4-8148-11e7-91dd-50beb8b93846.png)

Note that it visually points to the 3rd parameter, the empty string, very prominently.

However, the actual issue is that `logger`, the variable being declared is unused.

Emitting an incorrect column is an issue that is magnified in the case of editors that visualize that information more prominently than what's visible by glancing at terminal output only.

**Comment by shurcooL** [comment]
@griesemer This is a feature that's new to Go 1.9, and it has a bug. What's the reason this issue is added to Go 1.10 milestone instead of 1.9?

Is it because the bug is considered to be minor and happens only in a few cases, so the feature is still overall helpful and worth being included, despite the incorrect behavior in this particular case?

**Comment by griesemer** [comment]
@shurcooL It's not a show stopper. Yes, it's a new feature, and we know it has bugs (mostly due to the nature of how the existing code deals with position information). It can be disabled with the -C cmd/compile flag. It's definitively minor, even though it's annoying and perhaps confusing if one is using an IDE and not very familiar with the language. We will fix all these over time and the intention is to have 100% accurate positions.

**Comment by odeke-em** [comment]
Great news, seems like @mdempsky fixed this issue with commit f22ef70254a6a79834ed3fa21c4d1ccadeb758bd and CL https://go-review.googlesource.com/c/66810/
Running it on tip 32f994a gives me

```shell
$ go version && go run main.go 
go version devel +32f994a Mon Nov 6 07:09:50 2017 +0000 darwin/amd64
# command-line-arguments
./main.go:6:9: n declared and not used
./main.go:6:12: err declared and not used
```

I'll submit a test to ensure we don't regress.

**Comment by odeke-em** [comment]
Alright, addressed by CL https://go-review.googlesource.com/c/go/+/76150 and commit https://github.com/golang/go/commit/0c554957483aaf1f62c6251dbe62ab5fce3e219c.
